File: app/src/main/java/com/example/soroushprofile/models/ConversationFactory.java
package com.example.soroushprofile.models;

import com.annimon.stream.Optional;
import com.example.soroushprofile.R;

import java.util.ArrayList;
import java.util.List;

public final class ConversationFactory {

    public static ConversationThread getThread(ConversationType type) {
        switch (type) {
            case individual:
                String bio = "27 y.o. designer from tehran, iran..., The Biotechnology Innovation Organization\n" +
                        " is the largest trade organization in the world that represents the biotechnology industry.";
                User user = new User("Farhad Azad", R.drawable.p02, Optional.of(bio));
                List<String> media = new ArrayList<String>() {{
                    add("1.jpg");
                    add("2.jpg");
                }};
                return new IndividualConversation(Optional.of(media), user);
            case group:
                return new GroupConversation(Optional.empty(), Optional.empty(), "Soroush", 231);
            case channel:
                return new ChannelConversation(Optional.empty(), Optional.of("khabar 24 is live.."), "Khabar24", 429);
            default:
                throw new IllegalArgumentException("not supported type.");
        }
    }

}

File: app/src/main/java/com/example/soroushprofile/models/ConversationFactory.kt
package com.example.soroushprofile.models

import com.example.soroushprofile.R

object ConversationFactory {

    fun getThread(type: ConversationType): ConversationThread {
        when (type) {
            ConversationType.Individual -> {
                val bio = "27 y.o. designer from tehran, iran..., The Biotechnology Innovation Organization" +
                        " is the largest trade organization in the world that represents the biotechnology industry."
                val user = User("Farhad Azad", R.drawable.header, bio)
                val prefix = "https://baelm.net/wp-content/uploads/2014/11"
                val media = arrayListOf(
                        "$prefix/Nature-Photo-2-634x467.jpg",
                        "$prefix/Nature-Photo-3.jpg",
                        "$prefix/Nature-Photo-5-634x475.jpg",
                        "$prefix/Nature-Photo-101-634x475.jpg",
                        "$prefix/Nature-Photo-14-634x396.jpg",
                        "$prefix/Nature-Photo-16-634x410.jpg",
                        "$prefix/Nature-Photo-19-634x396.jpg",
                        "$prefix/Nature-Photo-2-634x467.jpg",
                        "$prefix/Nature-Photo-3.jpg",
                        "$prefix/Nature-Photo-5-634x475.jpg",
                        "$prefix/Nature-Photo-101-634x475.jpg",
                        "$prefix/Nature-Photo-14-634x396.jpg",
                        "$prefix/Nature-Photo-16-634x410.jpg",
                        "$prefix/Nature-Photo-19-634x396.jpg"
                )
                return IndividualConversation(media, user)
            }

            ConversationType.Group -> return GroupConversation(null, null, "Soroush", 231)
            ConversationType.Channel -> return ChannelConversation(null, "khabar 24 is live..", "Khabar24", 429)
        }
    }

}
